lib/kubernetes-deploy/kubernetes_resource/cloudsql.rb

```diff
@@ -0,0 +1,28 @@
+module KubernetesDeploy
+  class Cloudsql < KubernetesResource
+    TIMEOUT = 30.seconds
+
+    def initialize(name, namespace, file)
+      @name, @namespace, @file = name, namespace, file
+    end
+
+    def sync
+      _, st = run_kubectl("get", type, @name)
+      @found = st.success?
+      @status = true
+      log_status
+    end
+
+    def deploy_succeeded?
+      exists?
```

Comment:
> Does deployment fail readiness until the instance is available, or could it? That would be great. Otherwise, still better than nothing. A resource looking for another resource isn't entirely unprecedented here, in that I've got services waiting to have endpoints before succeeding.

The Deployment (and thus Service) cannot be created until the instance is created.